.buildkite/filter-pipeline.py

```diff
@@ -0,0 +1,119 @@
+#! /usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+# This script performs filtering of a buildkite pipeline.yml configuration
+# file. The filter is based on the contents of 'only_changes' steps
+# attributes.
+# 'only_changes' attribute should be a list of regexes, changes in filepaths
+# that match one of these regexes, should trigger the given step to run.
+#
+# If the current step is triggered to run and it depends on another step,
+# the step we depend on is also triggered to run to provide the required
+# dependencies.
+# And vice versa, if the current step depends on the step which was triggered to run,
+# the current step is also triggered to run to use updated dependencies in
+# dependent steps.
+#
+# Consider the following example (lower steps depend on the higher):
+#       A
+#      / \
+#     B  C
+#    /   |
+#   D    E
+#   |   / \
+#   F  G  H
```

Comment:
One thing that is missing on this graph is that it's possible for a step to depend on multiple steps.

AFAICT the script does cover that, but I think it would be nice to put this in this example and cover it by a test as well, e.g. make `G` depend on `D` as well.